DataFormats/SiPixelClusterSoA/interface/alpaka/SiPixelClustersDevice.h

```diff
@@ -0,0 +1,55 @@
+#ifndef DataFormats_SiPixelClusterSoA_SiPixelClustersDevice_h
+#define DataFormats_SiPixelClusterSoA_SiPixelClustersDevice_h
+
+#include <cstdint>
+#include <alpaka/alpaka.hpp>
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "DataFormats/SiPixelClusterSoA/interface/SiPixelClustersHost.h"
+#include "DataFormats/Portable/interface/alpaka/PortableCollection.h"
+#include "DataFormats/SiPixelClusterSoA/interface/SiPixelClustersLayout.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/CopyToHost.h"
+
+// TODO: The class is created via inheritance of the PortableCollection.
+// This is generally discouraged, and should be done via composition.
+// See: https://github.com/cms-sw/cmssw/pull/40465#discussion_r1067364306
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  class SiPixelClustersDevice : public PortableCollection<SiPixelClustersLayout<>> {
```

Comment:
Just to remind, this approach doesn't work for host backends, as for them the `ALPAKA_ACCELERATOR_NAMESPACE::SiPixelClustersDevice` must be the same type as `SiPixelClustersHost`.